go/mysql/server_test.go

```diff
@@ -510,29 +505,26 @@ func TestConnCounts(t *testing.T) {
 	c, err := Connect(context.Background(), params)
 	require.NoError(t, err, "Connect failed")
 
-	connCounts := connCountPerUser.Counts()
-	assert.Equal(t, 1, len(connCounts)-initialNumUsers)
 	checkCountsForUser(t, user, 1)
 
 	// Test with a second new connection.
 	c2, err := Connect(context.Background(), params)
 	require.NoError(t, err)
-	connCounts = connCountPerUser.Counts()
-	// There is still only one new user.
-	assert.Equal(t, 1, len(connCounts)-initialNumUsers)
 	checkCountsForUser(t, user, 2)
 
-	// Test after closing connections. time.Sleep lets it work, but seems flakey.
+	// Test after closing connections.
 	c.Close()
-	// time.Sleep(10 * time.Millisecond)
-	// checkCountsForUser(t, user, 1)
+	assert.EventuallyWithT(t, func(t *assert.CollectT) {
+		checkCountsForUser(t, user, 1)
+	}, 1*time.Second, 10*time.Millisecond)
```

Comment:
There's a helper these days in `testify` that you can use to eventually get some state. It's probably useful in more places where we manually do something like this today :smile:.